### ISSUE TITLE
Tag Helpers: Ignore '@' symbols in Tag Helper attributes if not string or character.

### ISSUE THREAD

**Issue by NTaylorMullen** [aspnet/Razor#129]
The typical flow for a Razor developer is to write `@Foo` when they want to switch from HTML to C# code.  Now that we're dealing with HTML attributes that could potentially be projected into a C# context we need to consider the impact.

Do we ignore `@` symbols in C# projected HTML attributes?

AKA:
```C#
public class CurrentAgeTagHelper : TagHelper
{
    public int Age { get; set; }
    ...
}
```

```
@{
    myAge = "3";
}

<currentAge age="@myAge" />
<currentAge age="myAge" /> @* This has equivalent output to the line above *@
<currentAge age="3" /> @* This has equivalent output to the line above *@
```

**Comment by dougbu** [comment]
will fix together with https://github.com/aspnet/Mvc/issues/1253

**Comment by dougbu** [comment]
Fixed with 26afdbd88959c94c17755f1ae7ce4add4fbe8489 through de4cafa8cdbe357ca3c4a34b084e3cf09c0db231
